op/op.go

```diff
@@ -61,6 +61,14 @@ func New(id string, result map[string]any, status Status, err error, params map[
 	}
 }
 
+func NewCancel(id string, err error, params map[string]any, start time.Time) Op {
```

Comment:
Minor: To help prep for library extract, it may be worth adding a doc comment to this and `NewTimeout` so we don't have to go back and add them later.